app/controllers/jobseekers/job_applications_controller.rb

```diff
@@ -26,12 +25,17 @@ def new
   end
 
   def create
-    new_job_application = current_jobseeker.job_applications.create(vacancy:)
-    redirect_to jobseekers_job_application_apply_path(new_job_application)
+    new_job_application = if vacancy.has_uploaded_form?
+                            current_jobseeker.uploaded_job_applications.create!(vacancy:)
+                          else
+                            current_jobseeker.native_job_applications.create!(vacancy:)
```

Comment:
we could just have `new_job_application = current_jobseeker.job_applications.create!(vacancy:)`  and add a `before_create`  callback on the base model that would set the correct `type`

Reply:
I ended up encapsulating this in vacancy, just because i find callbacks make things a bit more difficult to debug and understand for me personally!